lib/Backend/arm64/LowerMD.cpp

```diff
@@ -6008,73 +5958,70 @@ LowererMD::GenerateClz(IR::Instr * instr)
+        // cbz tmp2, done
+        // s1 = JavascriptNumber::k_Nan
+        // done:
+
+        IR::RegOpnd* tmp = IR::RegOpnd::New(TyMachReg, m_func);
+        IR::Instr* newInstr = IR::Instr::New(Js::OpCode::MVN, tmp, s1, m_func);
         instrInsert->InsertBefore(newInstr);
-        LegalizeMD::LegalizeInstr(newInstr, false);
+        LowererMD::Legalize(newInstr);
 
-        symVTableDst = IR::SymOpnd::New(tempNumberSym, TyMachPtr, this->m_func);
-        symDblDst = IR::SymOpnd::New(tempNumberSym, (uint32)Js::JavascriptNumber::GetValueOffset(), TyFloat64, this->m_func);
-        symTypeDst = IR::SymOpnd::New(tempNumberSym, (uint32)Js::JavascriptNumber::GetOffsetOfType(), TyMachPtr, this->m_func);
+        newInstr = IR::Instr::New(Js::OpCode::AND, tmp, tmp, IR::AddrOpnd::New((Js::Var)0x7FF0000000000000, IR::AddrOpndKindConstantVar, m_func, true), m_func);
+        instrInsert->InsertBefore(newInstr);
 
-        if (this->m_lowerer->outerMostLoopLabel == nullptr)
-        {
-            // If we are not in loop, just insert in place
-            numberInitInsertInstr = instrInsert;
-        }
-        else
-        {
-            // Otherwise, initialize in the outer most loop top if we haven't initialize it yet.
-            numberInitInsertInstr = this->m_lowerer->initializedTempSym->TestAndSet(tempNumberSym->m_id) ?
-                nullptr : this->m_lowerer->outerMostLoopLabel;
-        }
-    }
-    else
-    {
-        this->GenerateNumberAllocation(dstOpnd, instrInsert, isHelper);
-        symVTableDst = IR::IndirOpnd::New(dstOpnd, 0, TyMachPtr, this->m_func);
-        symDblDst = IR::IndirOpnd::New(dstOpnd, (uint32)Js::JavascriptNumber::GetValueOffset(), TyFloat64, this->m_func);
-        symTypeDst = IR::IndirOpnd::New(dstOpnd, (uint32)Js::JavascriptNumber::GetOffsetOfType(), TyMachPtr, this->m_func);
-        numberInitInsertInstr = instrInsert;
-    }
+        IR::LabelInstr* helper = Lowerer::InsertLabel(true, instrInsert);
+        IR::Instr* branch = IR::BranchInstr::New(Js::OpCode::CBZ, helper, m_func);
+        branch->SetSrc1(tmp);
+        helper->InsertBefore(branch);
 
-    if (numberInitInsertInstr)
-    {
-        IR::Opnd *jsNumberVTable = m_lowerer->LoadVTableValueOpnd(numberInitInsertInstr, VTableValue::VtableJavascriptNumber);
+        IR::LabelInstr* done = Lowerer::InsertLabel(isHelper, instrInsert);
+
+        Lowerer::InsertBranch(Js::OpCode::Br, done, helper);
+
+        IR::RegOpnd* tmp2 = IR::RegOpnd::New(TyMachReg, m_func);
 
-        // STR dst->vtable, JavascriptNumber::vtable
-        newInstr = IR::Instr::New(Js::OpCode::STR, symVTableDst, jsNumberVTable, this->m_func);
-        numberInitInsertInstr->InsertBefore(newInstr);
-        LegalizeMD::LegalizeInstr(newInstr, false);
+        newInstr = IR::Instr::New(Js::OpCode::AND, tmp2, s1, IR::AddrOpnd::New((Js::Var)0x000FFFFFFFFFFFFFull, IR::AddrOpndKindConstantVar, m_func, true), m_func);
+        done->InsertBefore(newInstr);
```

Comment:
Better here to use UBFX dst, src, #0, #52 which is guaranteed to be a single instruction (parameters would by tmp2, s1, IR::AddrOpnd::New(BITFIELD(0, 52), ...)